### ISSUE TITLE
SwinV2 for downstream

### ISSUE THREAD

**Issue by sipie800** [open-mmlab/mmpretrain#1862]
### Branch

main branch (mmpretrain version)

### Describe the bug

I tried to adapt swinv2 in mmpretrain to dino in mmdet.
The config is swinv2 tiny and dino and I use the provided converted weight. Just change the backbone.
Train on coco. The issue is that it won't converge. As soon as the first log output, the cls loss is about 1.3 or higher. And the grad_norm is nan. After a few steps, the grad_norm get a value but the loss go even higher. And after about 5000 steps, all loss go very large number. Finaly the 1st epoch got all 0.0 mAPs.

As long as I rewrite only the backbone cfg into swin tiny(mmdet one or mmpretrain one) or any backbones, the 1st step cls loss is less than 1.1, which means that that's the normal value. And 1st epoch got 0.29 around mAP. So the bug is in mmpretrain swinv2.

I've tried many tuned hyp like lr or arch and hear nothing.
There must be something wrong in your implementation of swinv2. What I‘ve found so far is that you mean to re-init the relative position embedding in a downtream task right? But actually there is no any relative position embedding at all, while the position embedding in pretrained weight is deleted indeed. Unless we use absolute position embedding but that's not the way.

### Environment

mmdet 3 dev, mmpretrain 1.2

**Comment by sipie800** [comment]
Well the thing is fp16 training. If one should get a weird underfiting problem in swinv2, turn off amp.
